### PR TITLE
feat: AVS upgrade for correct reporting on Eigenlayer UI

### PR DESCRIPTION
**Pull request by aroralanuk**
### Description

- upgraded the AVS on holesky - now working splendidly!
- included the function call for upgrading the AVS on mainnet `upgradeHsm4090`
    - encoded calldata to `0x9e3045a8Feb10F05f4d1268Ef8E303b655e8F562` is `0x99a88ec4000000000000000000000000e8e59c6c8b56f2c178f63bcfc4ce5e5e2359c8fc0000000000000000000000009e3045a8feb10f05f4d1268ef8e303b655e8f562 `
    - make the call `forge script script/avs/DeployAVS.s.sol:DeployAVS --sig "upgradeHsm4090(string memory,address,address)" "ethereum" 0xe8E59c6C8B56F2c178f63BCFC4ce5e5e2359c8fc 0x272CF0BB70D3B4f79414E0823B426d2EaFd48910 --rpc-url https://rpc.ankr.com/eth -vvvv --sender 0x3965AC3D295641E452E0ea896a086A9cD7C6C5b6` to repro
    

### Drive-by changes

Found two issues with forge scripting 
- upgrade call wasn't working with a pranked sender but the equivalent low-level call worked
- once you call `vm.expectRevert()` the script terminates.

### Related issues

- fixes https://github.com/hyperlane-xyz/issues/issues/1297

### Backward compatibility

Yes

### Testing

Fork run of upgrade

### CODE REVIEW

File: solidity/script/avs/DeployAVS.s.sol [hunk 1]
```diff
@@ -35,6 +36,8 @@ contract DeployAVS is Script {
     Quorum quorum;
     uint256 thresholdWeight = 6667;
 
+    address KILN_OPERATOR_ADDRESS = 0x1f8C8b1d78d01bCc42ebdd34Fae60181bD697662;
```
Comment (yorhodes) [inline solidity/script/avs/DeployAVS.s.sol:39]: is there an easy way to test this for more operators?

File: solidity/script/avs/DeployAVS.s.sol
Comment (yorhodes) [inline solidity/script/avs/DeployAVS.s.sol:1]: `vm.prank` and `vm.expectRevert` are only applicable in unit tests iiuc
for testing the xERC20 limits on a persistent anvil node, I used the anvil RPCs
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/dcd6dc502e9f0d34d0b6e860cbafb12e3e01d1c8/solidity/script/xerc20/GrantLimits.s.sol#L23-L26
I think you are right to use the `--sender` and `--fork-url` flags

File: solidity/script/avs/DeployAVS.s.sol [hunk 2]
```diff
@@ -168,4 +171,59 @@ contract DeployAVS is Script {
...
+            (
+                ITransparentUpgradeableProxy(payable(hsmProxy)),
+                address(strategyManagerImpl)
+            )
+        );
+        console.log("Encoded upgrade call: ");
+        console.logBytes(encodedUpgradeCalldata);
+
+        // only meant for simulating the call on mainnet as the actual caller needs to the gnosis safe
+        address(proxyAdmin).call(encodedUpgradeCalldata);
```
[the remark this answers is not in the document]
Comment (yorhodes) [reply solidity/script/avs/DeployAVS.s.sol:218]: https://book.getfoundry.sh/tutorials/best-practices#scripts

**Review by yorhodes — APPROVED** [review]
would like to avoid manually editing the implementation address in the calldata log

File: solidity/script/avs/DeployAVS.s.sol (hunk 1 above)
Comment (aroralanuk) [reply solidity/script/avs/DeployAVS.s.sol:39]: There's a mapping in the AVSDirectory which I can query and put it in the json file but I felt like this was enough.

**Pull request merged**